server/src/core/import_resolver.rs

```diff
@@ -434,64 +442,189 @@ fn _resolve_new_symbol(session: &mut SessionInfo, parent: Rc<RefCell<Symbol>>, n
     return Err("Symbol not found".to_string())
 }
 
-pub fn get_all_valid_names(session: &mut SessionInfo, source_file_symbol: &Rc<RefCell<Symbol>>, from_stmt: Option<&Identifier>, base_name: String, level: Option<u32>) -> HashSet<OYarn> {
-    //A: search base of different imports
+/*
+Used for autocompletion. Given a base_name, return all valid names that can be used to complete it.
+is_from indicates if the import is the X in "from X import Y". Else it is Y from "import Y" or "from X import Y"
+*/
+pub fn get_all_valid_names(session: &mut SessionInfo, source_file_symbol: &Rc<RefCell<Symbol>>, from_stmt: Option<String>, import: String, level: Option<u32>, is_from: bool) -> HashMap<OYarn, SymType> {
+    let (identifier_from, to_complete) = match from_stmt {
+        Some(from_stmt_inner) => {
+            if is_from {
+                let split = from_stmt_inner.split(".").collect::<Vec<&str>>();
+                if split.len() > 1 {
+                    (Some(Identifier::new(split[0..split.len()-1].join(".").as_str(), TextRange::default())), split.last().unwrap().to_string())
+                } else {
+                    (None, split.last().unwrap().to_string())
+                }
+            } else {
+                (Some(Identifier::new(from_stmt_inner.clone(), TextRange::default())), import.clone())
+            }
+        },
+        None => (None, S!("")),
```

Comment:
Shouldn't this be `import.clone()` ? 

Like that `import odo|` will try to list everything because it is starting from empty string? 
```suggestion
        None => (None, import.clone()),
```